trio/_core/_context.py

```diff
@@ -0,0 +1,24 @@
+from contextvars import Context
+
+from async_generator import asynccontextmanager
+
+from . import _run
+
+
+@asynccontextmanager
+async def change_context(context):
+    """Asynchronous context manager to change the :class:`contextvars.Context`
+    for the current task.
+
+    """
+    task = _run.current_task()
+    saved_context = task.context
+    task.context = context
+
+    try:
+        await _run.checkpoint()
+        yield
+    finally:
+        task.context = saved_context
+        # I assume this is right?
```

Comment:
The second one is cancel-shielded because it would produce a second cancellation event if the stuff inside was cancelled. The first one isn't because it seems better to not have to do two round trips through the scheduler if the task is already cancelled.

Reply:
That's fair -- I guess anyone who wants to change the context without taking a cancellation can also use a shielded scope. I think most permutations here are defensible as long as they're documented.